server/src/test/java/com/cloud/vm/UserVmManagerTest.java

```diff
@@ -630,9 +631,9 @@ public void testUpdateVmNicIpSuccess2() throws Exception {
         when(vlan.getVlanNetmask()).thenReturn("255.255.255.0");
 
         when(_ipAddrMgr.allocatePublicIpForGuestNic(Mockito.eq(_networkMock), nullable(Long.class), Mockito.eq(_accountMock), anyString())).thenReturn("10.10.10.10");
-        lenient().when(_ipAddressDao.findByIpAndSourceNetworkId(anyLong(), anyString())).thenReturn(null);
+        when(_ipAddressDao.findByIpAndSourceNetworkId(anyLong(), eq("10.10.10.10"))).thenReturn(newIp);
+        when(_ipAddressDao.findByIpAndSourceNetworkId(anyLong(), eq("10.10.10.9"))).thenReturn(null);
```

Comment:
Sorry I hope I'm wrong here, but isn't this too specific, looks like hardcoded to me?

Reply:
@borisstoyanov yes, it is hardcoded, but it is just an unit test.